apis/r/src/arrow.cpp

```diff
@@ -118,14 +119,26 @@ void createSchemaFromArrow(const std::string& uri, naxpSchema nasp, naxpArray na
         Rcpp::stop(tfm::format("Error: Array '%s' already exists", uri));
     }
 
-    // create the ArraySchema
-    auto as = tdbs::ArrowAdapter::tiledb_schema_from_arrow_schema(ctx, std::move(schema),
-                                                                  std::pair(std::move(dimarr),
-                                                                            std::move(dimsch)),
-                                                                  datatype, sparse,
-                                                                  pltcfg);
-    // Create the schema at the given URI
-    tiledb::Array::create(uri, as);
+    if (datatype == "SOMADataFrame") {
+        tdbs::SOMADataFrame::create(uri, std::move(schema),
+                                    std::pair(std::move(dimarr), std::move(dimsch)),
+                                    sctx, pltcfg);
+    } else if (datatype == "SOMASparseNDArray") {
+        // for arrays n_children will be three as we have two dims and a data col
+        std::string datacoltype = sp->children[sp->n_children-1]->format;
+        tdbs::SOMASparseNDArray::create(uri, datacoltype,
+                                        std::pair(std::move(dimarr), std::move(dimsch)),
+                                        sctx, pltcfg);
+    } else if (datatype == "SOMADenseNDArray") {
+        // for arrays n_children will be three as we have two dims and a data col
+        std::string datacoltype = sp->children[sp->n_children-1]->format;
+        tdbs::SOMADenseNDArray::create(uri, datacoltype,
+                                       std::pair(std::move(dimarr), std::move(dimsch)),
+                                       sctx, pltcfg);
```

Comment:
Not necessary to do, but an alternative to doing these if-else statements would be to create a `std::map<std::string, std::function<...>>` where the string datatype maps to the associated `create` static method.

Reply:
Yes I adopted that in one or two places too and like it.  The explicit if/else if/else trees allow for logging and better debugging though.